src/ipa_pc/mod.rs

```diff
@@ -111,21 +111,43 @@ impl<G: AffineCurve, D: Digest> InnerProductArgPC<G, D> {
         for (labeled_commitment, value) in labeled_commitments.zip(values) {
             let commitment = labeled_commitment.commitment();
             combined_v += &(cur_challenge * &value);
-            combined_commitment_proj += &labeled_commitment.commitment().comm.mul(cur_challenge);
+
+            if commitment.comm.len() > 1 {
+
+                let mut comm_lc = <G::Projective as ProjectiveCurve>::zero();
+                for (i, comm_single) in commitment.comm.iter().enumerate() {
+                    let is = i * n;
+                    comm_lc += &comm_single.mul(point.pow(&[is as u64]));
+                }
+                combined_commitment_proj += &comm_lc.mul(&cur_challenge);
+
+            } else {
```

Comment:
I think that the code in the if clause works also for the case that commitment.com.len() = 1? If so, is there a stylistic reason do the distinction?

Reply:
If you execute the code in the if clause you perform additional (but useless) operation, with respect to the else clause. The result will be the same but you waste time and resources

Reply:
Well, it is minimal extra code that is executed. In my opinion it would be better style without separations into cases.

Reply:
This is already redone